### ISSUE TITLE
Add a compatibility shim for `commands.getstatusoutput`?

### ISSUE THREAD

**Issue by ncoghlan** [benjaminp/six#207]
`six` currently includes a not-yet-released change to add `six.moves` entries for `commands.getoutput` and `commands.getstatusoutput`: https://github.com/benjaminp/six/commit/e05c7253043647e37bdbc13038bff06334fac8cb

While this is correct for `commands.getoutput`,  `subprocess.getstatusoutput` unfortunately broke backwards compatibility with Python 2's `commands.getstatusoutput` back in 3.3.4 (see https://bugs.python.org/issue22635 for details), which means the Python 3 API needs a compatibility shim to shift the exit code left by 8 bits to get it back into the format produced by `os.wait()`: https://docs.python.org/3/library/os.html#os.wait

Accordingly, a backwards compatible `getstatusoutput` implementation on Py3 actually looks like this:

    def py2_getstatusoutput(cmd):
        status, output = subprocess.getstatusoutput(command)
        if os.name != 'nt':
            # convert status to be interpreted according to the wait() rules
            status = status << 8
        return status, output

I'm not sure how to incorporate that into the `six.moves` structure, though :(

**Comment by benjaminp** [comment]
That's certainly unfortunate. I suppose for starters, I will remove `getstatusoutput` from `six.moves`.

**Comment by ncoghlan** [comment]
Perhaps the simplest fix would be to offer a top level `six.getstatusoutput` function?

The main downside I can see to that is that it would lead to the following odd discrepancy:

    from six.moves import getoutput
    from six import getstatusoutput

Since it's only one function, `getoutput` could also be made top-level, rather than having one of them top-level and the other under `moves`.

**Comment by benjaminp** [comment]
Yeah, it's not beautiful but there's already strange things like `six.assertCountEqual`.

**Comment by StilesCrisis** [comment]
Did anyone ever make progress on fixing this? AFAICS it looks like the broken version was taken out last year, but no one ever merged back in the fixed version.
